### PR TITLE
Describe EigenGate._eigen_components better

### PR DESCRIPTION
**Pull request by Strilanc**

### CODE REVIEW

File: cirq/ops/eigen_gate.py [hunk 1]
```diff
@@ -64,23 +82,57 @@ def _with_exponent(self: TSelf,
...
+            The Pauli Z gate's eigencomponents are:
+
+                [
+                    (0, np.array([[1, 0],
+                                  [0, 0]])),
+                    (1, np.array([[0, 0],
+                                  [0, 1]])),
+                ]
+
+            Valid eigencomponents for Rz(π) = iZ are:
```
Comment (kevinsung) [inline cirq/ops/eigen_gate.py]: Rz(π) = exp(-i π Z / 2) is actually -iZ.

Comment (Strilanc) [reply cirq/ops/eigen_gate.py]: Oops. Fixed.

File: cirq/ops/eigen_gate.py [hunk 2]
```diff
@@ -23,6 +23,24 @@
...
+        # The θ in λ = exp(i π θ) where λ is a unique eigenvalue. The exponent
+        # factor is used, instead of just a raw unit complex number, because it
+        # disambiguates several cases. For example, when λ=-1 you can set θ to
+        # -1 instead of +1 resulting in square root operations returning -i
+        # instead of +1.
+        ('eigenvalue_exponent_factor', float),
+
+        # The projection matrix onto the eigenspace of the eigenvalue. Must
+        # equal Σ_k |λ_k⟩⟨λ_k| where the |λ_k⟩ vectors form an orthonormal
+        # basis for the eigenspace.
```
Comment (dabacon) [inline cirq/ops/eigen_gate.py]: This one could probably do with an explicitly worked out example vector.

Comment (dabacon) [reply cirq/ops/eigen_gate.py]: Oh see you did this below.  OK.

**Pull request merged**